src/core/internal/traits.d

```diff
@@ -164,6 +164,26 @@ template isInnerClass(T) if (is(T == class))
         enum isInnerClass = false;
 }
 
+/// Detect whether type `T` is a static array.
+enum isStaticArray(T) = is(T : U[N], U, size_t N);
```

Comment:
I think the only question that remains is whether we want to allow implicit conversions, i.e. types which have an `alias this` to a static array. Or `enum`s whose base type is a static array:
https://run.dlang.io/gist/run-dlang/e18c94ccff92c2d09feba048b7700860?compiler=dmd

Reply:
> I think the only question that remains is whether we want to allow implicit conversions, i.e. types which have an alias this to a static array

I say *no*.  We want to know if it *is* a static array not whether or not it implicitly converts to a static array or behaves like a static array.

> Or enums whose base type is a static array:
I say *yes*.  That is an enumeration of static arrays.  The enum items are indeed static arrays.

The implementation has been updated and test cases have been added.  Nice work identifying these edge cases!